_data/devops-checklist.yml

```diff
@@ -0,0 +1,690 @@
+        Your apps may need different configuration settings in each environment: e.g., different memory settings,
+        different features on or off. Define these in config files that get checked into version control (e.g.,
+        dev-config.yml, stage-config.yml, prod-config.yml) and packaged with your app artifact (i.e., packaged directly
+        into the Docker image for your app), and have your app boot up code pick the proper config file for the current
+        environment during boot.
+
+    - title: Define your infrastructure as code
+      description: |
+        Do not deploy anything by hand, by using the AWS Console, or the AWS CLI. Instead, define all of your
+        <a href="https://blog.gruntwork.io/a-comprehensive-guide-to-terraform-b3d32832baca" target="_blank">infrastructure as code</a>
+        using tools such as <a href="https://www.terraform.io/" target="_blank">Terraform</a>,
+        <a href="https://www.packer.io/" target="_blank">Packer</a>, and
+        <a href="https://www.docker.com/" target="_blank">Docker</a>.
+
+    - title: Test your infrastructure code
+      description: |
+        If all of your infrastructure is defined as code, you can create automated tests for it. The goal is to verify
+        your infrastructure works as expected after every single commit, long before those infrastructure changes
+        affect prod. See <a href="https://github.com/gruntwork-io/terratest" target="_blank">Terratest</a> for more
+        info.
+
+    - title: Set up immutable infrastructure
+      description: |
+        Don't update EC2 Instance or Docker containers in place. Instead, launch completely new EC2 Instances and new
+        Docker containers and, once those are up and healthy, remove the old EC2 Instances and Docker images. Since we
+        never "modify" anything, but simply replace, this is known as
+        <a href="https://www.oreilly.com/ideas/an-introduction-to-immutable-infrastructure" target="_blank">immutable infrastructure</a>,
+        and it makes it easier to reason about what's deployed and to manage that infrastructure.
+
+    - title: Promote artifacts
+      description: |
+        Deploy immutable artifacts to one environment at a time, and promote it to the next environment after testing.
+        For example, you might deploy v0.3.2 to dev, and test it there. If it works well, you promote the exact same
+        artifact, v0.3.2, to stage, and test it there. If all goes well, you finally promote v0.3.2 to prod. Since it's
+        the exact same code in every environment, there's a good chance that if it works in one environment, it'll
+        also work in the others.
+
+    - title: Roll back in case of failure
+      description: |
+        If you use immutable, versioned artifacts as your unit of deployment, then any time something goes wrong, you
+        have the option to roll back to a known-good state by rolling out the previous version. If your infrastructure
```

Comment:
by rolling _back_ the previous version?

Reply:
Fixed! https://github.com/gruntwork-io/gruntwork-io.github.io/pull/45/commits/ac969266fa45512affe5434f70111b488f8c4eb8